### PR TITLE
Improved support for AVX512.

### PR DESCRIPTION
**Pull request by benoitsteiner**
Added missing definition for the Packet16q16i.
Fixed a couple of bugs in the implementation of max reductions for
avx512

### CODE REVIEW

File: third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/PacketMathAVX512.h [hunk 1]
```diff
@@ -457,7 +457,7 @@ EIGEN_STRONG_INLINE QInt16 predux_max<Packet32q16i>(const Packet32q16i& a) {
   std::uint32_t w =
       pfirst(
         _mm_max_epi16(res, _mm_shuffle_epi32(res, _MM_SHUFFLE(0, 0, 0, 1))));
-  return std::min({
+  return std::max({
```
Comment (drpngx) [inline third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/PacketMathAVX512.h]: Just double checking here that it's correct?

**Comment by drpngx** [comment]
```
/workspace/tensorflow/contrib/makefile/downloads/eigen/unsupported/Eigen/CXX11/../../../Eigen/src/Core/GenericPacketMath.h:291:10: error: no match for 'operator=' (operand types are 'Eigen::QInt16' and 'const Eigen::internal::Packet16q16i')
 {  (*to) = from; }
```

**Comment by drpngx** [comment]
Jenkins, test this please.

**Review by drpngx — APPROVED** [review]

**Comment by drpngx** [comment]
Ready to go, right?

**Comment by benoitsteiner** [comment]
It's ready.

**Pull request merged**